dlrover/python/master/elastic_training/rdzv_manager.py

```diff
@@ -95,11 +95,18 @@ def remove_alive_node(self, node: Node):
         """When a node is exited, the master will remove it from alive list."""
         if node.id in self._alive_nodes:
             self._alive_nodes.remove(node.id)
+            self._has_node_failed = True
```

Comment:
Agree with the logic of this line. However, I am concerned that it may affect other parts, as previous decisions were based on node rank.

Reply:
The logic to build rendezvous based on node rank does not change in the PR. The PR only add the node id in the `NodeTopologyMeta` and print `node_id` in logs.